### PR TITLE
Forms and Data Handling: Use single GET request and req.query for Search functionality

### LINKED ISSUE #28585: Node: Update "Forms and Data Handling" lesson to use a single GET request for search functionality
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

The current assignment in the "Forms and Data Handling" lesson suggests implementing a `/search` route that accepts both GET and POST requests:
image
This approach is not ideal for a search functionality, as searching is a data retrieval operation, it is idempotent and does not contain sensitive data. Therefore, this functionality should be implemented with a single GET request.

The GET request can be implemented with a form GET method. The data will be passed as a query string. We can then access this data in express js through `req.query`.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling

### (Optional) Discord Name

Cake

### PR DESCRIPTION
**Pull request by DrantDumani**
## Because
This PR is meant to close issue #28585 which explains that POST is not ideal for search requests and that a GET request should be used instead.

## This PR
- The last paragraph of [HTML Forms Overview](https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling#html-forms-overview) states that data sent via a GET request is sent as a query string in the url itself.
- The search method in [Implement Searching](https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling#implement-searching) says to only use a GET request. 
- It is explicitly stated that data sent via a GET request will not be available in `req.body` and that `req.query` should be used instead.

## Issue
Closes #28585 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Looks like Prettier jumped in with its formatting, which does things very differently to our own style guide. Before I can review further, please could to undo all of the formatting changes and ensure you don't run Prettier on files in this repo? If you have it running on save, you should get very familiar with "save without formatting" or some alternative to formatting on save, as not every project will be compatible with it. This includes the `/` in the input tag on line 24 which is a classic Prettier behaviour (and is actually against the HTML5 spec).

File: nodeJS/express/forms_and_data_handling.md [hunk 1]
```diff
@@ -541,11 +541,10 @@ Don't forget to update the view to display these new fields!
...
 What if we want to search for a specific user in a list of thousands? We'll need a new route and view that lets clients search our list of users.
 
-1. Add a form (in `createUser.ejs` or another view) which accepts a `name` or `email` (or both!)
-1. Create a new route `/search` which accepts `GET` and `POST` requests.
-1. Add the search logic to your controller which searches your list for a matching user.
-   - Your `POST` request should handle searching for the user.
-   - Your `GET` request should then render the search result.
+1. Add a form with a `GET` method (in `createUser.ejs` or another view) which accepts a `name` or `email` (or both!)
+1. Create a new route `/search` which accepts a `GET` request.
+1. Add the search logic to your controller which searches your list for a matching user. Form data that has been sent via a `GET` request will not be available in the `req.body`. You will need to use `req.query` instead.
```
Comment (MaoShizhong) [inline nodeJS/express/forms_and_data_handling.md:546]: Nit
```suggestion
1. Add the search logic to your controller which searches your list for a matching user. Form data that has been sent via a `GET` request will not be available via `req.body`. You will need to use `req.query` instead.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
While we're at it, as another nit, the rogue extra space at the end of the opening form tag on line 22 can be removed as it shouldn't be there. Prettier did pick it up but came with a lot more than bargained for :stuck_out_tongue:

File: nodeJS/express/forms_and_data_handling.md (hunk 1 above)
Comment (DrantDumani) [reply nodeJS/express/forms_and_data_handling.md]: Done! The extra space is gone and `in the` has been changed to `via`.

**Review by MaoShizhong — APPROVED** [review]
:pray:

**Pull request merged**